pkg/tracing/tracing.go

```diff
@@ -17,8 +18,16 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+func ShutdownExporter(ctx context.Context, exp *otlptrace.Exporter) {
```

Comment:
Shutting down the exporter is the wrong level of granularity. You should shut down the `TracerProvider` instead, which takes care of waiting on span processors and shutting them down as necessary.

Reply:
I should remove the ShutdownExporter functions from both metrics and traces then? I have already added the ShutdownController function which shuts down the metric provider and trace provider respectively..

Reply:
Yes, I would remove these functions. An exporter gets registered with a `TracerProvider` as a `SpanProcessor`; the `TracerProvider` effectively takes ownership of the exporter's lifecycle, so there's no reason to shut the exporter down separately unless you create it but fail to register it with a `TracerProvider`, and need to clean it up.

Note that this wasn't always so in the Go SDK. It took a few months of gestation before it got easier to clean all of this up.

Reply:
Ok, got it 👍